hack/generated/controllers/controller_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	storage "github.com/Azure/k8s-infra/hack/generated/apis/microsoft.storage/v20190401"
+	"github.com/Azure/k8s-infra/hack/generated/pkg/armclient"
+	"github.com/Azure/k8s-infra/hack/generated/pkg/genruntime"
+)
+
+func Test_ResourceGroup_CRUD(t *testing.T) {
+	// TODO: Can we put this someplace common?
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	g := NewGomegaWithT(t)
+	ctx := context.Background()
+
+	// Create a resource group
+	rg := testContext.NewTestResourceGroup()
+	err := testContext.KubeClient.Create(ctx, rg)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// It should be created in Kubernetes
+	g.Eventually(testContext.Ensure.ProvisioningComplete(ctx, rg)).Should(BeTrue())
+	g.Expect(testContext.Ensure.ProvisioningSuccessful(ctx, rg)).Should(BeTrue())
+
+	g.Expect(rg.Status.Location).To(Equal(testContext.AzureRegion))
+	g.Expect(rg.Status.Properties.ProvisioningState).To(Equal(string(armclient.SucceededProvisioningState)))
+	g.Expect(rg.Status.ID).ToNot(BeNil())
+	armId := rg.Status.ID
+
+	// Delete the resource group
+	err = testContext.KubeClient.Delete(ctx, rg)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Eventually(testContext.Ensure.Deleted(ctx, rg)).Should(BeTrue())
+
+	// Ensure that the resource group was really deleted in Azure
+	// TODO: Do we want to just use an SDK here? This process is quite icky as is...
```

Comment:
This is really icky -- we need a better way to do it; unclear to me if that's fixing our internal interface to make this cleaner or just using an SDK.

If using an SDK we'll end up with a ton of SDKs which is sorta a downside.